src/porepy/numerics/fv/biot.py

```diff
@@ -608,10 +611,344 @@ def compute_stress(self, g, u, data):
+                either a mono-dimensional or a mixed-dimensional problem.
+            d (dictionary): Data dictionary associated with the grid. Not used,
+                but included for consistency reasons.
+        Returns:
+            np.array (g.num_cells): Pressure solution vector. Will be identical
+                to solution_array.
+        """
+        return solution_array
+
+    def discretize(self, g, data):
+        """ Discretize the pressure gradient term of the Biot equation.
+
+        Parameters:
+            g (pp.Grid): grid, or a subclass, with geometry fields computed.
+            data (dict): For entries, see above.
+
+        Raises:
+            NotImplementedError, the discretization should be performed using the
+            discretize method of the Biot class.
+        """
+        raise NotImplementedError(
+            """No discretize method implemented for the GradP
+                                  class. See the Biot class."""
+        )
+
+    def assemble_matrix_rhs(self, g, data):
+        """ Return the matrix and right-hand side for a discretization of the pressure
+        gradient term of the Biot equation.
+
+        Parameters:
+            g : grid, or a subclass, with geometry fields computed.
+            data: dictionary to store the data. For details on necessary keywords,
+                see method discretize()
+
+        Returns:
+            matrix: sparse csr (g.dim * g_num_cells, g.dim * g_num_cells) Discretization
+            matrix.
+            rhs: array (g.dim * g_num_cells) Right-hand side.
+        """
+        return self.assemble_matrix(g, data), self.assemble_rhs(g, data)
+
+    def assemble_matrix(self, g, data):
+        """ Return the matrix and right-hand side for a discretization of the pressure
+        gradient term of the Biot equation.
+
+        Parameters:
+            g (Grid): Computational grid, with geometry fields computed.
+            data (dictionary): With data stored.
+
+        Returns:
+            scipy.sparse.csr_matrix: System matrix of this discretization. The
+                size of the matrix will depend on the specific discretization.
+
+        Raises:
+            ValueError if the pressure gradient term has not already been discretized.
+        """
+        matrix_dictionary = data[pp.DISCRETIZATION_MATRICES][self.keyword]
+        if not "grad_p" in matrix_dictionary:
+            raise ValueError(
+                """GradP class requires a pre-computed discretization to be
+                             stored in the matrix dictionary."""
+            )
+        biot_alpha = data[pp.PARAMETERS][self.keyword]["biot_alpha"]
+        return matrix_dictionary["grad_p"] * biot_alpha
+
+    def assemble_rhs(self, g, data):
+        """ Return the zero right-hand side for a discretization of the pressure
+        gradient term.
+
+        @Runar: Is it correct that this is zero.
```

Comment:
Probably not

Reply:
It will have a contribution from boundary terms, similar to standard mpfa/mpsa. This may require some thinking.